tests/dependency_audit_tests.cdc

```diff
@@ -0,0 +1,217 @@
+        "../transactions/migration-contract-staging/stage_contract.cdc",
+        ["A", aUpdateCadence],
+        aAccount
+    )
+    Test.expect(aStagingTxResult, Test.beSucceeded())
+
+    events = Test.eventsOfType(Type<DependencyAudit.UnstagedDependencies>())
+        Test.assertEqual(2, events.length)
+
+    commitResult = executeTransaction(
+        "../transactions/dependency-audit/admin/test_check_dependencies.cdc",
+        [addresses, names, authorizers],
+        adminAccount
+    )
+    Test.expect(commitResult, Test.beSucceeded())
+
+    events = Test.eventsOfType(Type<DependencyAudit.UnstagedDependencies>())
+        Test.assertEqual(2, events.length)
+}
+
+
+access(all) fun testChekDependenciesWithMixedEntries() {
+    var events = Test.eventsOfType(Type<DependencyAudit.UnstagedDependencies>())
+    Test.assertEqual(2, events.length)
+
+    let addresses: [Address] = [adminAccount.address, fooAccount.address, aAccount.address, bcAccount.address]
+    let names: [String] = ["DependencyAudit", "Foo", "A", "B"]
+    let authorizers: [Address] = []
+    var commitResult = executeTransaction(
+        "../transactions/dependency-audit/admin/test_check_dependencies.cdc",
+        [addresses, names, authorizers],
+        adminAccount
+    )
+    Test.expect(commitResult, Test.beSucceeded())
+
+    events = Test.eventsOfType(Type<DependencyAudit.UnstagedDependencies>())
+        Test.assertEqual(3, events.length)
+
+    let evt = events[2] as! DependencyAudit.UnstagedDependencies
+    Test.assertEqual(2, evt.dependencies.length)
+    Test.assertEqual(fooAccount.address, evt.dependencies[0].address)
+    Test.assertEqual("Foo", evt.dependencies[0].name)
+    Test.assertEqual(bcAccount.address, evt.dependencies[1].address)
+    Test.assertEqual("B", evt.dependencies[1].name)
+}
+
+access(all) fun testSetPanic() {
+    let shouldPanic: Bool = true
+    var commitResult = executeTransaction(
+        "../transactions/dependency-audit/admin/set_unstaged_cause_panic.cdc",
+        [shouldPanic],
+        adminAccount
+    )
+    Test.expect(commitResult, Test.beSucceeded())
+
+    var events = Test.eventsOfType(Type<DependencyAudit.PanicOnUnstagedDependenciesChanged>())
+        Test.assertEqual(1, events.length)
+
+    let evt = events[0] as! DependencyAudit.PanicOnUnstagedDependenciesChanged
+    Test.assertEqual(true, evt.shouldPanic)
+}
+
+
+access(all) fun testChekDependenciesWithUnstagedEntriesPanics() {
+    let addresses: [Address] = [fooAccount.address]
+    let names: [String] = ["Foo"]
+    let authorizers: [Address] = []
+    let commitResult = executeTransaction(
+        "../transactions/dependency-audit/admin/test_check_dependencies.cdc",
+        [addresses, names, authorizers],
+        adminAccount
+    )
+    Test.expect(commitResult, Test.beFailed())
+    // not sure how to test this:
+    // Test.expect(commitResult.error!.message, Test.contain("panic: Found unstaged dependencies: A.0x0000000000000008.Foo") )
```

Comment:
I wasn't able to find docs or prior art on how to implement and pass a custom handler to `Test.expect`, but I also validated locally that this is the reason the txn fails here

Reply:
Thank you!